### PR TITLE
Print out debug Wininet errors

### LINKED ISSUE #5720: Print wininet error message in debug spew
can we get the message in addition to the hresult? i.e. construct a hresult_error from it and then use the .message?

_Originally posted by @asklar in https://github.com/microsoft/react-native-windows/pull/5610_

### PR DESCRIPTION
**Pull request by asklar**
I commented about this in PR #5610 and followed up offline. Wininet errors strings are not normally resolved by cppwinrt since they require knowing they are wininet errors and cppwinrt doesn't do that: https://github.com/microsoft/cppwinrt/issues/710

This change adds the code to the web socket module to print out the right string, instead of "The text associated with this error code could not be found".

Fixes #5720

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/5717)

### CODE REVIEW

**Comment by NickGerleman** [comment]
Separately, the new websocket code @junielkatarn wrote has seen mass usage in production scenarios. We should try to converge to its implementation since IIRC it was more feature complete than the existing UWP module.

**Comment by jonthysell** [comment]
@asklar Are there other modules like this that need special methods to get error strings in cppwinrt, and is this something that can be consolidated into reusable code?

Part of my work for async native module call examples was to capture error strings for unhandled exceptions and still resolve promises: https://github.com/microsoft/react-native-windows-samples/blob/7ee179393ecf424ed2de4d36d32bf0ade3f75e53/samples/NativeModuleSample/cppwinrt/windows/NativeModuleSample/AsyncMethodExamples.h#L61

In react-native-camera I took it a step further with a template:

https://github.com/react-native-community/react-native-camera/blob/d7092048ff01cf9cdf7baa9b4314c7bb9810a52d/windows/ReactNativeCameraCPP/ReactCameraModule.h#L56

I planned on adding something like that to Microsoft.ReactNative.Cxx for external developers to use, so if there's a method we could write that was smart about looking up error strings that would be great.

**Comment by asklar** [comment]
> @asklar Are there other modules like this that need special methods to get error strings in cppwinrt, and is this something that can be consolidated into reusable code?
> 
> Part of my work for async native module call examples was to capture error strings for unhandled exceptions and still resolve promises: https://github.com/microsoft/react-native-windows-samples/blob/7ee179393ecf424ed2de4d36d32bf0ade3f75e53/samples/NativeModuleSample/cppwinrt/windows/NativeModuleSample/AsyncMethodExamples.h#L61
> 
> In react-native-camera I took it a step further with a template:
> 
> https://github.com/react-native-community/react-native-camera/blob/d7092048ff01cf9cdf7baa9b4314c7bb9810a52d/windows/ReactNativeCameraCPP/ReactCameraModule.h#L56
> 
> I planned on adding something like that to Microsoft.ReactNative.Cxx for external developers to use, so if there's a method we could write that was smart about looking up error strings that would be great.

@jonthysell it seems like wininet is the odd one out here. DLLs can define their own error messages and have clients of it call FormatMessage if they determine they are messages from that DLL, but in reality few people do this and they just rely on system error messages. I tried playing with system_error on godbolt but couldn't get it to work, is this an msvc extension of some sort?

File: vnext/Microsoft.ReactNative/Modules/WebSocketModuleUwp.cpp [hunk 1]
```diff
@@ -164,7 +165,25 @@ void WebSocketModule::WebSocket::connect(
 
   if (!SUCCEEDED(hr)) {
     winrt::hresult_error e{hr};
-    OutputDebugString("WebSocket.connect failed (0x%8X) %ls\n", e);
+    std::wstring error;
+#define INET_ERR_OUT_FORMAT_BUFFER_SIZE 256
+    const DWORD err = hr & 0xffff;
+    if (err >= INTERNET_ERROR_BASE && err <= INTERNET_ERROR_LAST) {
+      wchar_t errorStr[INET_ERR_OUT_FORMAT_BUFFER_SIZE] = {};
```
Comment (JunielKatarn) [inline vnext/Microsoft.ReactNative/Modules/WebSocketModuleUwp.cpp:172]: Is this change related to suppressing exception in favor of HRESULT?
Does this code retrieve the error message originally provided with the exception?

**Comment by JunielKatarn** [comment]
> Separately, the new websocket code @JunielKatarn wrote has seen mass usage in production scenarios. We should try to converge to its implementation since IIRC it was more feature complete than the existing UWP module.

There has been discussion about this with @acoates-ms. That's the mid-term plan.
There are a couple of tasks on the "Desktop" implementation I want to take care of before this, to prevent any regressions on UWP.

I'll open an issue to track this.

**Pull request merged**